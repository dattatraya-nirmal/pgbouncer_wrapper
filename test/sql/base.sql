\set ECHO 0
BEGIN;
\i sql/pgbouncer_wrapper.sql
\set ECHO all

-- You should write your tests

SELECT pgbouncer_wrapper('foo', 'bar');

SELECT 'foo' #? 'bar' AS arrowop;

CREATE TABLE ab (
    a_field pgbouncer_wrapper
);

INSERT INTO ab VALUES('foo' #? 'bar');
SELECT (a_field).a, (a_field).b FROM ab;

SELECT (pgbouncer_wrapper('foo', 'bar')).a;
SELECT (pgbouncer_wrapper('foo', 'bar')).b;

SELECT ('foo' #? 'bar').a;
SELECT ('foo' #? 'bar').b;

ROLLBACK;
