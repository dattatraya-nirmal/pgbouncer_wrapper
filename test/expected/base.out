\set ECHO none
-- Make sure that the views are actually there.
SELECT * FROM pgbouncer.active_sockets WHERE false;
 type | user | database | state | addr | port | local_addr | local_port | connect_time | request_time | ptr | link | remote_pid | recv_pos | pkt_pos | pkt_remain | send_pos | send_remain | pkt_avail | send_avail 
------+------+----------+-------+------+------+------------+------------+--------------+--------------+-----+------+------------+----------+---------+------------+----------+-------------+-----------+------------
(0 rows)

SELECT * FROM pgbouncer.clients WHERE false;
 type | user | database | state | addr | port | local_addr | local_port | connect_time | request_time | ptr | link | remote_pid 
------+------+----------+-------+------+------+------------+------------+--------------+--------------+-----+------+------------
(0 rows)

SELECT * FROM pgbouncer.config WHERE false;
 key | value | changeable 
-----+-------+------------
(0 rows)

SELECT * FROM pgbouncer.databases WHERE false;
 name | host | port | database | force_user | pool_size | reserve_pool | pool_mode | max_connections | current_connections 
------+------+------+----------+------------+-----------+--------------+-----------+-----------------+---------------------
(0 rows)

SELECT * FROM pgbouncer.dns_hosts WHERE false;
 hostname | ttl | addrs 
----------+-----+-------
(0 rows)

SELECT * FROM pgbouncer.dns_zones WHERE false;
 zonename | serial | count 
----------+--------+-------
(0 rows)

SELECT * FROM pgbouncer.fds WHERE false;
 fd | task | user | database | addr | port | cancel | link | client_encoding | std_strings | datestyle | timezone | password 
----+------+------+----------+------+------+--------+------+-----------------+-------------+-----------+----------+----------
(0 rows)

SELECT * FROM pgbouncer.lists WHERE false;
 list | items 
------+-------
(0 rows)

SELECT * FROM pgbouncer.mem WHERE false;
 name | size | used | free | memtotal 
------+------+------+------+----------
(0 rows)

SELECT * FROM pgbouncer.pools WHERE false;
 database | user | cl_active | cl_waiting | sv_active | sv_idle | sv_used | sv_tested | sv_login | maxwait | pool_mode 
----------+------+-----------+------------+-----------+---------+---------+-----------+----------+---------+-----------
(0 rows)

SELECT * FROM pgbouncer.servers WHERE false;
 type | user | database | state | addr | port | local_addr | local_port | connect_time | request_time | ptr | link | remote_pid 
------+------+----------+-------+------+------+------------+------------+--------------+--------------+-----+------+------------
(0 rows)

SELECT * FROM pgbouncer.sockets WHERE false;
 type | user | database | state | addr | port | local_addr | local_port | connect_time | request_time | ptr | link | remote_pid | recv_pos | pkt_pos | pkt_remain | send_pos | send_remain | pkt_avail | send_avail 
------+------+----------+-------+------+------+------------+------------+--------------+--------------+-----+------+------------+----------+---------+------------+----------+-------------+-----------+------------
(0 rows)

SELECT * FROM pgbouncer.stats WHERE false;
 database | total_requests | total_received | total_sent | total_query_time | avg_req | avg_recv | avg_sent | avg_query 
----------+----------------+----------------+------------+------------------+---------+----------+----------+-----------
(0 rows)

SELECT * FROM pgbouncer.users WHERE false;
 name | pool_mode 
------+-----------
(0 rows)

ROLLBACK;
